tls/s2n_connection.c

```diff
@@ -78,6 +82,11 @@ struct s2n_connection *s2n_connection_new(s2n_mode mode)
     conn->managed_io = 0;
     conn->corked_io = 0;
 
+    /* By default, only the client will authenticate the Server's Certificate. The Server does not request or
+     * authenticate any client certificates. */
+    conn->client_cert_auth_type = S2N_CERT_AUTH_NONE;
+    conn->verify_cert_chain_callback = deny_all_certs;
```

Comment:
I know that we already have `S2N_ENABLE_CLIENT_MODE` and require that to be enabled before doing unsafe things on the Client Side, but this callback will now also be called on the server side during Mutual Auth to verify clients, so relying on `S2N_ENABLE_CLIENT_MODE` may not be enough anymore. The Server side code of s2n should always be secure by default. 

How do you feel about adding a `S2N_ENABLE_UNSAFE_CERT_VALIDATION` or renaming `S2N_ENABLE_CLIENT_MODE`?